server/utils.py

```diff
@@ -236,6 +236,16 @@ def chunks(l, n):
         prev_index = index
 
 
+def output_csv_iterable(header, rows):
```

Comment:
This is too similar to `generate_csv` below to be another function. It has other problems: it uses Windows line endings, and will buffer the whole thing in memory (which isn't really an issue for these jobs, but it defeats the while purpose of using iterables).

Instead, try to use `generate_csv`. It would looks something like
```
    csv = generate_csv(subms, header, get_row)
```
where `subms` is the thing you're iterating over, and `get_row` is a function that returns a dict of CSV values.